non-fungible-token/src/lib.rs

```diff
@@ -0,0 +1,214 @@
+#![no_std]
+#![feature(const_btree_new)]
+
+use codec::{Decode, Encode};
+use gstd::{exec, msg, prelude::*, ActorId};
+pub mod base;
+use base::NonFungibleTokenBase;
+pub mod token;
+use token::TokenMetadata;
+
+use primitive_types::{H256, U256};
+use scale_info::TypeInfo;
+
+const GAS_RESERVE: u64 = 500_000_000;
+const ZERO_ID: ActorId = ActorId::new([0u8; 32]);
+
+#[derive(Debug)]
+pub struct NonFungibleToken {
+    pub name: String,
+    pub symbol: String,
+    pub base_uri: String,
+    pub owner_by_id: BTreeMap<U256, ActorId>,
+    pub token_metadata_by_id: BTreeMap<U256, TokenMetadata>,
+    pub token_approvals: BTreeMap<U256, ActorId>,
+    pub balances: BTreeMap<ActorId, U256>,
+    pub operator_approval: BTreeMap<ActorId, BTreeMap<ActorId, bool>>,
```

Comment:
```suggestion
    pub operator_approval: BTreeMap<ActorId, BTreeSet<ActorId>>,
```
if for `A` set contains `B` then it is approved operator. Otherwise not.

Reply:
Yeah, you are right